src/scalib_ext/ranklib/src/histogram.rs

```diff
@@ -7,11 +7,17 @@ mod bnp {
     include!(concat!(env!("OUT_DIR"), "/binding_bnp.rs"));
 }
 
+const CONV_LIMIT: f64 = 34359738368.0 as f64; //define a certain limit
+
 pub trait Histogram {
     fn new(size: usize) -> Self;
+    fn type_str(&self) -> &'static str;
```

Comment:
I see you're using this string to check the type of the histogram. This would be better solved using an Enum: https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html